src/pipeline-stages/src/main/scala/UDFTransformer.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.{ParamMap, UDFParam}
+import org.apache.spark.ml.util.{ComplexParamsReadable, ComplexParamsWritable, Identifiable}
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+object UDFTransformer extends ComplexParamsReadable[UDFTransformer]
+
+/** <code>UDFTransformer</code> takes as input input column, output column, and a UserDefinedFunction
+  * returns a dataframe comprised of the original columns with the output column as the result of the
+  * udf applied to the input column
+  */
+class UDFTransformer(val uid: String) extends Transformer with Wrappable with ComplexParamsWritable
+  with HasInputCol with HasOutputCol {
+  def this() = this(Identifiable.randomUID("UDFTransformer"))
+
+  val udf = new UDFParam(this, "udf", "User defined function to be applied to the DF input col")
+
+  /** @group getParam */
+  def getUDF: UserDefinedFunction = $(udf)
+
+  /** @group setParam */
+  def setUDF(value: UserDefinedFunction): this.type = set("udf", value)
+
+  /** @param dataset - The input dataset, to be transformed
+    * @return The DataFrame that results from applying the udf to the inputted dataset
+    */
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    transformSchema(dataset.schema, logging = true)
```

Comment:
unless there's a side effect I'm not noticing, this line doesn't do anything right?

Reply:
HasInputCol? It sets up the inoutCol and outputCol params.

Reply:
are we looking at the same lines? I'm seeing

`transformSchema(dataset.schema, logging = true)`

Reply:
This is done to check the schema early as a fail fast.